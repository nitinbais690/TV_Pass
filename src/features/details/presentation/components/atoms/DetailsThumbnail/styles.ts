import { StyleSheet } from 'react-native';
import { AspectRatio, scale } from 'qp-common-ui';
import { isTablet } from 'core/styles/AppStyles';

export const imageAspectRatio = AspectRatio._2by3;

export const styles = () => {
    return StyleSheet.create({
        container: {
            ...(isTablet
                ? {
                      width: scale(100),
                      height: scale(116.13),
                  }
                : {
                      width: scale(62),
                      height: scale(72),
                  }),
        },
        imageStyle: {
            flex: 1,
            width: '100%',
            aspectRatio: imageAspectRatio,
            borderRadius: scale(8),
        },
    });
};
