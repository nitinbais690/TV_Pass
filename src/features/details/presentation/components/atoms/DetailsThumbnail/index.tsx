import React from 'react';
import { View, Platform } from 'react-native';
import { ResizableImage } from 'qp-discovery-ui';
import { ImageType } from 'qp-common-ui';
import { useDimensions } from '@react-native-community/hooks';
import { imageAspectRatio, styles } from './styles';

export default function DetailsThumbnail(props: DetailsPosterProps) {
    const { width, height } = useDimensions().window;
    const isPortrait = height > width;
    const posterStyles = styles();

    return (
        <View style={posterStyles.container}>
            {!Platform.isTV && (
                <ResizableImage
                    keyId={props.id || ''}
                    style={[posterStyles.imageStyle]}
                    imageType={ImageType.Banner}
                    aspectRatioKey={imageAspectRatio}
                    skipResize={false}
                    isPortrait={isPortrait}
                    source={{ uri: props.imageUrl }}
                />
            )}
        </View>
    );
}

interface DetailsPosterProps {
    mainResourceImage: string | undefined;
    imageUrl: string | undefined;
    id: string | undefined;
}
