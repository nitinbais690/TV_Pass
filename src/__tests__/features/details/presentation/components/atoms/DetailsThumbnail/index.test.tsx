import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import DetailsThumbnail from 'features/details/presentation/components/atoms/DetailsThumbnail';

jest.mock('core/presentation/hooks/use-app-colors');

describe('detailsThumbnail', () => {
    const props = {
        mainResourceImage: 'assets/images/video_placeholder.png',
        id: '01234',
        imageUrl: 'assets/images/got.png',
    };

    test('renders correctly', () => {
        const { container } = render(
            <DetailsThumbnail mainResourceImage={props.mainResourceImage} id={props.id} imageUrl={props.imageUrl} />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });
});
