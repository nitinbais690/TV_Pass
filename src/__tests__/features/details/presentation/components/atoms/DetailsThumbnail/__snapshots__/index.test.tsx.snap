// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailsThumbnail renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 104.91428571428571,
        "width": 90.34285714285714,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "aspectRatio": 0.6666666666666666,
              "borderRadius": 11.657142857142857,
              "flex": 1,
              "width": "100%",
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "cache": "web",
            "uri": "assets/images/got.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
